### PR TITLE
Minor fix for running "mlflow deployments create -t triton --flavor triton ..."

### PR DESCRIPTION
**Pull request by yeahdongcn**
UnboundLocalError: local variable 'meta_dict' referenced before assignment

The above error shows in listing models in Triton model repository

### CODE REVIEW

File: deploy/mlflow-triton-plugin/mlflow_triton/deployments.py [hunk 1]
```diff
@@ -193,6 +193,8 @@ def list_deployments(self):
                         )['Body'].read().decode('utf-8'))
                 elif os.path.isfile(mlflow_meta_path):
                     meta_dict = self._get_mlflow_meta_dict(d['name'])
+                else:
+                    continue
```
Comment (GuanLuo) [inline deploy/mlflow-triton-plugin/mlflow_triton/deployments.py:197]: Can you share an example that the "reference before assignment" error occurs? I wonder if this change will shadow the real cause of the issue.

This code block is executed when the model is ready in Triton server, the error seems to be happening only if the Triton server is also used for other deployments, i.e. some models being served is not added via MLFlow plugin and thus they don't have `meta_dict` stored in model folder.

If that is the case, then this change makes sense to filter only the MLFlow deployed models.

Comment (yeahdongcn) [reply deploy/mlflow-triton-plugin/mlflow_triton/deployments.py:197]: In my case, besides the models published from MLFlow, there are some other models in my model repository (host path). These models don't contain `meta.yaml` so `meta_dict` won't be declared in the context.

Comment (GuanLuo) [reply deploy/mlflow-triton-plugin/mlflow_triton/deployments.py:197]: That makes sense. Then it looks fine that the model not deployed via MLFlow should be agnostic to the plugin user.

**Review by GuanLuo — APPROVED** [review]
LGTM. Will add test around this.

**Pull request merged**